cachito/workers/tasks/gitsubmodule.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+import tarfile
+import tempfile
+
+import git
+
+from cachito.errors import CachitoError
+from cachito.workers.pkg_managers.general import update_request_with_package
+from cachito.workers.tasks.celery import app
+from cachito.workers.paths import SourcesDir
+from cachito.workers.scm import Git
+
+__all__ = ["add_git_submodules_as_package"]
+log = logging.getLogger(__name__)
+
+
+@app.task
+def add_git_submodules_as_package(request_id, url, ref):
+    """
+    Add git submodules as package to the Cachtio request.
+
+    :param int request_id: the Cachito request ID this is for
+    :param str url: the source control URL to pull the source from
+    :param str ref: the source control reference
+    :raises CachitoError: if adding submodules as a package fail.
+    """
+    scm = Git(url, ref)
+    sources_dir = SourcesDir(scm.repo_name, f"{scm.ref}-with-submodules")
+    archive_path = sources_dir.archive_path
+    with tempfile.TemporaryDirectory(prefix="cachito-") as temp_dir:
+        with tarfile.open(archive_path, mode="r:gz") as tar:
+            tar.extractall(temp_dir)
```

Comment:
I'm looking at this code in `fetch_app_source()`: https://github.com/release-engineering/cachito/blob/58fe40c3710265fd0fc2c9113b1341425ff59685/cachito/workers/tasks/general.py#L46-L51
Maybe we don't even need to extract the source archive? The repo might be accessible like this:
```python
bundle_dir = RequestBundleDir(request_id)
repo = git.Repo(str(bundle_dir.source_root_dir))
```

Reply:
TBH I was also thinking about this yesterday but fixing unit tests took up too much time and I forgot to try this way. Will try and update if that works.